crates/solver/src/settlement_submission.rs

```diff
@@ -9,23 +9,79 @@ use anyhow::{anyhow, Result};
 use contracts::GPv2Settlement;
 use ethcontract::{
     errors::{ExecutionError, MethodError},
-    Account, TransactionHash,
+    Account, Address, TransactionHash,
 };
 use futures::FutureExt;
-use gas_estimation::GasPriceEstimating;
+use gas_estimation::{EstimatedGasPrice, GasPriceEstimating};
 use primitive_types::{H256, U256};
 use shared::Web3;
 use std::{
-    sync::Arc,
+    collections::HashMap,
+    sync::{Arc, Mutex},
     time::{Duration, Instant},
 };
 use submitter::{
-    DisabledReason, Submitter, SubmitterGasPriceEstimator, SubmitterParams, TransactionSubmitting,
+    DisabledReason, Submitter, SubmitterGasPriceEstimator, SubmitterParams, TransactionHandle,
+    TransactionSubmitting,
 };
 use web3::types::TransactionReceipt;
 
 const ESTIMATE_GAS_LIMIT_FACTOR: f64 = 1.2;
 
+// Key (Address, U256) represents pair (sender, nonce)
+type SubTxPool = HashMap<(Address, U256), Vec<(TransactionHandle, EstimatedGasPrice)>>;
+type TxPool = Arc<Mutex<Vec<SubTxPool>>>;
+
+#[derive(Default, Clone)]
+pub struct GlobalTxPool {
+    pools: TxPool,
+}
```

Comment:
Currently the code around SubTxPoolRef does not make use of the other available pools (only accesses its own index) so it seems like we could get rid of the global reference. In a github comment iirc you wrote that this is because it is likely needed in the future which we should comment here in the code too.

Reply:
Makes sense. Added comment to `SubTxPoolRef` struct